server/src/infra/repositories/database.repository.ts

```diff
@@ -210,6 +210,25 @@ export class DatabaseRepository implements IDatabaseRepository {
     return res as R;
   }
 
+  async withTryLock<R>(lock: DatabaseLock, callback: () => Promise<R>): Promise<R> {
+    let res;
+    const queryRunner = this.dataSource.createQueryRunner();
+    try {
+      const lockAcquired = await this.tryLock(lock, queryRunner);
+      if (lockAcquired) {
+        res = await callback();
+      }
+    } finally {
+      try {
+        await this.releaseLock(lock, queryRunner);
```

Comment:
will this line fail if the lock was not actually acquired?

Reply:
Fixed in latest version

Reply:
I think Postgres will just give a warning but it won't throw